relay-server/src/actors/global_config.rs

```diff
@@ -0,0 +1,138 @@
+use std::sync::Arc;
+use std::time::Duration;
+
+use relay_dynamic_config::GlobalConfig;
+use relay_system::{Addr, AsyncResponse, FromMessage, Interface, Sender, Service};
+use tokio::sync::mpsc::{self, UnboundedSender};
+
+use crate::actors::processor::EnvelopeProcessor;
+use crate::actors::project_upstream::GetProjectStates;
+use crate::actors::upstream::{SendQuery, UpstreamRelay};
+
+/// Service implementing the [`GlobalConfiguration`] interface.
+///
+/// The service is responsible for fetching the global config and
+/// forwarding it to the services that require it, and for serving downstream relays.
+#[derive(Debug)]
+pub struct GlobalConfigurationService {
+    enabled: bool,
+    global_config: Arc<GlobalConfig>,
+    envelope_processor: Addr<EnvelopeProcessor>,
+    upstream: Addr<UpstreamRelay>,
+}
+
+/// Service interface for the [`GetGlobalConfig`] message.
+pub enum GlobalConfiguration {
+    /// Used to receive the most recently fetched global config.
+    GetGlobalConfig(Sender<Arc<GlobalConfig>>),
+}
+
+impl Interface for GlobalConfiguration {}
+
+/// The message for requesting the most recent global config from [`GlobalConfigurationService`].
+pub struct GetGlobalConfig;
+
+impl FromMessage<GetGlobalConfig> for GlobalConfiguration {
+    type Response = AsyncResponse<Arc<GlobalConfig>>;
+
+    fn from_message(_: GetGlobalConfig, sender: Sender<Arc<GlobalConfig>>) -> Self {
+        Self::GetGlobalConfig(sender)
+    }
+}
+
+impl GlobalConfigurationService {
+    /// Creates a new [`GlobalConfigurationService`].
+    pub fn new(
+        enabled: bool,
+        envelope_processor: Addr<EnvelopeProcessor>,
+        upstream: Addr<UpstreamRelay>,
+    ) -> Self {
+        let global_config = Arc::new(GlobalConfig::default());
```

Comment:
How does the service behave when a global config is yet not received (i.e. we still have the default global config here), and Relay gets a request from downstream to fetch the global config?